src/libsyntax/ast.rs

```diff
@@ -68,6 +68,12 @@ pub struct Path {
     pub segments: Vec<PathSegment>,
 }
 
+impl PartialEq<Symbol> for Path {
+    fn eq(&self, symbol: &Symbol) -> bool {
+        self.segments.len() == 1 && self.segments[0].ident.name.interned() == *symbol
```

Comment:
There should be no `interned()` here.

Reply:
So these identifiers can never be gensymed?

Reply:
In theory they can and then we need to compare symbols rather than strings (i.e. no `interned()`).

In practice, all the gensyms are created by the compiler and I don't think we create gensyms for attribute/meta-item paths, for which this `PartialEq` happens to be is used.

Reply:
The path below does compare strings (i.e. ignores gensyms). I just copied the behavior from that.

Reply:
I added some debug assertions to ensure there's no gensyms here.